rayon-core/src/broadcast/mod.rs

```diff
@@ -99,13 +100,22 @@ where
     OP: Fn(BroadcastContext<'_>) -> R + Sync,
     R: Send,
 {
+    let current_thread = WorkerThread::current();
+    let current_thread_addr = current_thread as usize;
```

Comment:
use `expose_provenance` instead to clearly signal intent

Reply:
or use an AtomicPtr instead of an AtomicUsize for the latch. That should work, too, right?